src/main/java/am/ik/yavi/builder/ValidatorBuilder.java

```diff
@@ -732,6 +732,33 @@ public ValidatorBuilder<T> _doubleArray(ToDoubleArray<T> f, String name,
 		return this.constraint(f, name, c, DoubleArrayConstraint::new);
 	}
 
+	/**
+	 * @since 0.14.0
+	 */
+	public <C extends T> ValidatorBuilder<T> constraintOnClass(Class<C> clazz,
+			Validator<C> cValidator) {
+		Validator<T> TValidator = new ValidatorBuilder<T>()
+				.nest(clazz::cast, clazz.getName(), cValidator).build();
+
+		return constraintOnCondition(getClassConstraintCondition(clazz), TValidator);
+	}
+
+	/**
+	 * @since 0.14.0
+	 */
```

Comment:
These methods are quite advanced. Can you please add to the JavaDoc what use cases it is useful for and some sample code?

Reply:
absolutely